Terminal.Gui/ConsoleDrivers/CursesDriver/CursesDriver.cs

```diff
@@ -419,8 +419,22 @@ void ProcessInput (Action<KeyEvent> keyHandler, Action<KeyEvent> keyDownHandler,
 					mouseHandler (ToDriverMouse (ev));
 					return;
 				}
-				keyHandler (new KeyEvent (MapCursesKey (wch), keyModifiers));
-				keyUpHandler (new KeyEvent (MapCursesKey (wch), keyModifiers));
+				k = MapCursesKey (wch);
```

Comment:
What does all this do? It seems very fragile, just using hard coded numbers.

Reply:
@tig most of the code was changed because the shortcut keys compatibility as much keystrokes has not working. That code is for the functions keys combinations (F1-F12). In the middle I needed to fixes some bugs to test all the necessary code.

Reply:
> 
> 
> What does all this do? It seems very fragile, just using hard coded numbers.

As you know the wch variable is always mapped and converted to `Key `enum before they are passed to a event handler. I only assigned it before if no modifiers keys are necessary to shifted.